hadoop-crypto/src/main/java/com/palantir/crypto2/hadoop/EncryptedFileSystem.java

```diff
@@ -98,6 +100,23 @@ public boolean rename(Path src, Path dst) throws IOException {
         return renamed;
     }
 
+    @Override
+    public boolean delete(Path path, boolean recursive) throws IOException {
```

Comment:
could this just be
```
fs.delete(path, recursive);
keyStore.remove(path.toString());
```
That way the delete call takes care of figuring out if it's a directory or file.